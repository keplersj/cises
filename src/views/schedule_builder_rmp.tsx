import React from "react";
import styled from "@emotion/styled";

interface Props {
  rmpId: number;
  rating: number;
  tags: string[];
}

const Container = styled.div`
  border-radius: 15px;
  padding: 10px;
  background-color: #c0c0c0;
  color: #ffffff;
`;

const HeaderLink = styled.a`
  font-weight: 900;
  color: inherit;
  text-decoration: underline;

  :hover {
    color: inherit;
    text-decoration: underline;
  }

  :active {
    color: inherit;
    text-decoration: underline;
  }
`;

const TagContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

const Tag = styled.span`
  background-color: #a0a0a0;
  margin: 2px;
  padding-left: 1px;
  padding-right: 1px;
`;

export const ScheduleBuilderRMP = ({ rmpId, rating, tags }: Props) => (
  <Container id="cises-rmp">
    <HeaderLink
      href={`https://www.ratemyprofessors.com/ShowRatings.jsp?tid=${rmpId}`}
    >
      Info from Rate My Professors:
    </HeaderLink>
    <br />
    <strong>Average Rating: </strong>
    <span>{rating}</span>
    <br />
    <strong>Tags:</strong>
    <br />
    <TagContainer>
      {tags.map((tag, index) => (
        <Tag key={index}>{tag}</Tag>
      ))}
    </TagContainer>
  </Container>
);
