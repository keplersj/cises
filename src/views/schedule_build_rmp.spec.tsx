import React from "react";
import { ScheduleBuilderRMP } from "./schedule_builder_rmp";
import renderer from "react-test-renderer";

describe("Schedule Builder - Rate My Professor Component", () => {
  it("matches snapshot when provdided with valid data", () => {
    const component = renderer.create(
      <ScheduleBuilderRMP
        professor={{
          city_state_s: "",
          averageratingscore_rf: 2.0,
          pk_id: 2.0,
          schoolcity_s: "",
          schoolstate_full_s: "",
          pict_thumb_name_s: "",
          id: "",
          pageviews_i: 2.0,
          averagehelpfulscore_rf: 2.0,
          schoolcountry_s: "",
          schoolname_s: "",
          status_i: 2.0,
          teachermiddlename_t: "",
          averagehotscore_rf: 2.0,
          schoolstate_s: "",
          rated_date_dt: "",
          teacherfullname_s: "",
          teacherdepartment_s: "",
          total_number_of_ratings_i: 2.0,
          visible_i: 2.0,
          content_type_s: "",
          averageeasyscore_rf: 2.0,
          schoolid_s: "",
          teacherfirstname_t: "",
          teacherlastname_t: "",
          averageclarityscore_rf: 2.0,
          schoolwebpage_s: "",
          timestamp: "",
          tag_s_mv: ["", "", ""],
          tag_id_s_mv: ["", "", ""]
        }}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
