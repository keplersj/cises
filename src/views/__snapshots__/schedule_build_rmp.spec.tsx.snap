// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule Builder - Rate My Professor Component matches snapshot when provdided with valid data 1`] = `
.emotion-5 {
  border-radius: 15px;
  padding: 10px;
  background-color: #c0c0c0;
  color: #ffffff;
}

.emotion-0 {
  font-weight: 900;
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:hover {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:active {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.emotion-1 {
  background-color: #a0a0a0;
  margin: 2px;
  padding-left: 1px;
  padding-right: 1px;
}

<div
  className="emotion-5"
  id="cises-rmp"
>
  <a
    className="emotion-0"
    href="https://www.ratemyprofessors.com/ShowRatings.jsp?tid=2"
  >
    Info from Rate My Professors:
  </a>
  <br />
  <strong>
    Average Rating: 
  </strong>
  <span>
    2
  </span>
  <br />
  <strong>
    Tags:
  </strong>
  <br />
  <div
    className="emotion-4"
  >
    <span
      className="emotion-1"
    >
      
    </span>
    <span
      className="emotion-1"
    >
      
    </span>
    <span
      className="emotion-1"
    >
      
    </span>
  </div>
</div>
`;
